### PR TITLE
Bug 1775454: virtwhat multiple platform

### PR DESCRIPTION
**Pull request by pgier**
Fix cases where virtwhat returns multiple detected platforms.
Also provide a default path to TMPDIR in case the environment var isn't set.

### CODE REVIEW

File: text_collectors/virt.sh [hunk 1]
```diff
@@ -18,10 +18,10 @@ rm -f "${v}" || true
 touch "${v}"
 if [ -x /usr/sbin/virt-what ]
 then
-  platforms="$( virt-what )"
+  platforms=$(echo $( virt-what ) | tr '\n' ' ')
```
Comment (smarterclayton) [inline text_collectors/virt.sh]: Why is this necessary?  Testing locally, only removing the quotes around the ${platforms} is necessary.

Comment (smarterclayton) [reply text_collectors/virt.sh]: Also, how was this manifesting?  I wasn't seeing this in the metrics - 

```
count by (type) (max_over_time(cluster:virt_platform_nodes:sum[7d]))
```

was not showing an imbalance between `xen` and `xen-hvm` as I would expect if one of the tags was missing?

Comment (pgier) [reply text_collectors/virt.sh]: You're right, this part is probably not be necessary,  I wasn't sure if the bash for loop would correctly handle newlines.

In the related bug, the textfile collector is failing because of the invalid format of the generated file.  So it wasn't collecting the virt_platform metric at all for those nodes.
